diff --git a/javascript/contextMenus.js b/javascript/contextMenus.js
index ccae242f..55980d4e 100644
--- a/javascript/contextMenus.js
+++ b/javascript/contextMenus.js
@@ -136,16 +136,25 @@ var addContextMenuEventListener = initResponse[2];
     let generateOnRepeat = function(genbuttonid, interruptbuttonid) {
         let genbutton = gradioApp().querySelector(genbuttonid);
         let interruptbutton = gradioApp().querySelector(interruptbuttonid);
+        let coolDown = window.prompt("Seconds to cooldown", 30);
+		let coolDownCounter = coolDown;
         if (!interruptbutton.offsetParent) {
             genbutton.click();
         }
         clearInterval(window.generateOnRepeatInterval);
         window.generateOnRepeatInterval = setInterval(function() {
             if (!interruptbutton.offsetParent) {
-                genbutton.click();
+                if (coolDownCounter <= 0) {
+					genbutton.click();
+					coolDownCounter = coolDown;
+					genbutton.innerHTML = "Generate";
+		        } else {		        
+        		    coolDownCounter--;
+        		    genbutton.innerHTML = "Generate (Cooldown: "+coolDownCounter+")";
+        		}                
             }
         },
-        500);
+        1000);
     };
 
     let generateOnRepeat_txt2img = function() {
@@ -156,13 +165,15 @@ var addContextMenuEventListener = initResponse[2];
         generateOnRepeat('#img2img_generate', '#img2img_interrupt');
     };
 
-    appendContextMenuOption('#txt2img_generate', 'Generate forever', generateOnRepeat_txt2img);
-    appendContextMenuOption('#txt2img_interrupt', 'Generate forever', generateOnRepeat_txt2img);
-    appendContextMenuOption('#img2img_generate', 'Generate forever', generateOnRepeat_img2img);
-    appendContextMenuOption('#img2img_interrupt', 'Generate forever', generateOnRepeat_img2img);
+    appendContextMenuOption('#txt2img_generate', 'Generate forever with Cooldown', generateOnRepeat_txt2img);
+    appendContextMenuOption('#txt2img_interrupt', 'Generate forever with Cooldown', generateOnRepeat_txt2img);
+    appendContextMenuOption('#img2img_generate', 'Generate forever with Cooldown', generateOnRepeat_img2img);
+    appendContextMenuOption('#img2img_interrupt', 'Generate forever with Cooldown', generateOnRepeat_img2img);
 
     let cancelGenerateForever = function() {
         clearInterval(window.generateOnRepeatInterval);
+   		gradioApp().querySelector('#txt2img_generate').innerHTML = "Generate";
+		gradioApp().querySelector('#img2img_generate').innerHTML = "Generate";
     };
 
     appendContextMenuOption('#txt2img_interrupt', 'Cancel generate forever', cancelGenerateForever);
